File: Projeto_2/Projeto_2/teste/testeSeguro.js
import Carro from "../src/carro.js";
import Cliente from "../src/cliente.js";
import Seguro from "../src/seguro.js";

const testarSeguro = () => {
  const carro = new Carro(2020, "Toyota", "Corolla", "Branco", "XYZ-1234");
  const cliente = new Cliente("João Silva", "123.456.789-00");
  const seguro = new Seguro(carro, cliente, 1500, "2025-01-01");

  console.log("Testando Seguro:");
  console.log(seguro.calcularValor(1000, 0.15)); /*Espera: Valor calculado com base em 1000 + 15% (1150)*/
  console.log(seguro.verificarVigencia() ? "Seguro válido" : "Seguro expirado"); /*Espera: "Seguro válido"*/
};

testarSeguro();

File: Projeto_2/tests/teste_seguro.py
from datetime import date
from modelo import Modelo
from carro import Carro
from cliente import Cliente
from seguro import SeguroCarro

def exemplo_uso():
    modelo_corolla = Modelo("Corolla")
    carro1 = Carro(2020, "Toyota", modelo_corolla, "Branco", "XYZ-1234")
    carro2 = Carro(2018, "Toyota", Modelo("Etios"), "Prata", "ABC-9876")

    cliente1 = Cliente("João Silva", "123.456.789-00")

    seguro_carro = SeguroCarro(
        cliente=cliente1,
        carros=[carro1, carro2],
        valor=1500.00,
        vigencia_inicio=date(2024, 1, 1),
        vigencia_fim=date(2025, 1, 1)
    )

    print(seguro_carro)
    print("\nValor calculado:", seguro_carro.calcular_valor(1200.0, 0.1))
    print("Vigente?:", "Sim" if seguro_carro.verificar_vigencia() else "Não")

if __name__ == "__main__":
    exemplo_uso()
